### PR TITLE
Fixed objects within object ABI

### LINKED ISSUE #30: Objects within objects are not padded correctly and ABI problems come up
Returning object like:
```
{"abc", {"def", "ghi"}, "jkl"}
```
gives this output:
```
main() = {"abc", {"ghi", "jkl"}, "   "}
```

### PR DESCRIPTION
**Pull request by ViliamVadocz**
Also includes a temporary fix to optimisation by lowering the in-lining threshold and a rewrite of `sudoku.vlp`

### CODE REVIEW

**Comment by ViliamVadocz** [comment]
Resolves #30

File: volpe/compile.py [hunk 1]
```diff
@@ -61,15 +61,19 @@ def compile_and_run(llvm_ir, result_type, more_verbose=False, show_time=False, c
     target_machine.add_analysis_passes(pm)
 
     if more_verbose:
-        print("\nBefore optimization\n")
-        print(mod)
+        # print("\nBefore optimization\n")
```
Comment (LHolten) [inline volpe/compile.py:64]: no printing?

Comment (LHolten) [reply volpe/compile.py:64]: hmm, it is in fact easier to read as a file

**Comment by LHolten** [comment]
great

**Pull request merged**